pkg/scannerv4/mappers/mappers_test.go

```diff
@@ -899,6 +899,223 @@ func Test_toProtoV4Contents(t *testing.T) {
+					Issued:  now,
+					Updater: "unknown updater",
+				},
+			},
+			nvdVulns: map[string]map[string]*nvdschema.CVEAPIJSON20CVEItem{
+				"foo": {
+					"CVE-9999-999": {
+						ID: "CVE-9999-999",
+					},
+					"CVE-1234-567": {
+						ID: "CVE-1234-567",
+						Metrics: &nvdschema.CVEAPIJSON20CVEItemMetrics{
+							CvssMetricV31: []*nvdschema.CVEAPIJSON20CVSSV31{
+								{
+									CvssData: &nvdschema.CVSSV31{
+										Version:      "3.1",
+										VectorString: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			pkgEpss: map[string]epssDetail{
+				"CVE-1234-567": {
+					ModelVersion: "v2023.03.01",
+					CVE:          "CVE-1234-567",
+					Date:         "2025-01-15T00:00:00+0000",
+					EPSS:         0.00215,
+					Percentile:   0.59338,
+				},
+			},
+			want: map[string]*v4.VulnerabilityReport_Vulnerability{
+				"foo": {
+					EpssMetrics: &v4.VulnerabilityReport_Vulnerability_EPSS{
+						ModelVersion: "v2023.03.01",
+						Date:         "2025-01-15T00:00:00+0000",
+						Probability:  0.00215,
+						Percentile:   0.59338,
+					},
+					Id:     "foo",
+					Issued: protoNow,
+					Name:   "CVE-1234-567",
+					Cvss: &v4.VulnerabilityReport_Vulnerability_CVSS{
+						V3: &v4.VulnerabilityReport_Vulnerability_CVSS_V3{
+							Vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H",
+						},
+						Source: v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_NVD,
+						Url:    "https://nvd.nist.gov/vuln/detail/CVE-1234-567",
+					},
+					CvssMetrics: []*v4.VulnerabilityReport_Vulnerability_CVSS{
+						{
+							V3: &v4.VulnerabilityReport_Vulnerability_CVSS_V3{
+								Vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:H/A:H",
+							},
+							Source: v4.VulnerabilityReport_Vulnerability_CVSS_SOURCE_NVD,
+							Url:    "https://nvd.nist.gov/vuln/detail/CVE-1234-567",
+						},
+					},
+				},
+			},
+		},
+		"RHSA EPSS OK": {
+			ccVulnerabilities: map[string]*claircore.Vulnerability{
+				"foo": {
+					ID:      "foo",
+					Name:    "Name contains CVE-1234-567",
+					Issued:  now,
+					Updater: "rhel-vex",
+					Links:   "https://access.redhat.com/errata/RHSA-2021:1234",
+				},
+				"bar": {
+					ID:      "bar",
+					Name:    "Name contains CVE-7654-321",
+					Issued:  now,
+					Updater: "rhel-vex",
+					Links:   "https://access.redhat.com/errata/RHSA-2021:1234",
+				},
+			},
+			nvdVulns: nil,
+			pkgEpss: map[string]epssDetail{
+				"CVE-1234-567": {
+					ModelVersion: "v2023.03.01",
+					CVE:          "CVE-1234-567",
+					Date:         "2025-01-15T00:00:00+0000",
+					EPSS:         0.00215,
+					Percentile:   0.59338,
+				},
+				"CVE-7654-321": {
+					ModelVersion: "v2023.03.01",
+					CVE:          "CVE-7654-321",
+					Date:         "2025-01-15T00:00:00+0000",
+					EPSS:         0.04215,
+					Percentile:   0.69338,
+				},
+			},
+			want: map[string]*v4.VulnerabilityReport_Vulnerability{
+				"foo": {
```

Comment:
I think this will break after we merge the RHSA enricher. I think that work will merge these two entries if they are the same RHSA, regardless of their IDs.

Can you confirm if you're going to address the problem of CVEs for the same RHSA on this change? I am expecting [that work will have a mechanism for you to handle that](https://github.com/stackrox/stackrox/pull/13523#discussion_r1929378554) but I don't know if it's going to be part of this PR or future work. Can you please clarify?

Reply:
I see a TODO up above:

```
// TODO(ROX-27729): get the highest EPSS score for an RHSA across all CVEs associated with that RHSA
```

Maybe that TODO should come here as a reminder we need to update these tests.